### PR TITLE
Remove assertion and add test

### PR DESCRIPTION
**Pull request by Pratyush**
## Description

Removed an assertion and added some testing for an edge case.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (master)
- [ ] Linked to GitHub issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests
- [ ] Updated relevant documentation in the code
- [ ] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer

### CODE REVIEW

**Comment by mmagician** [comment]
This should solve the fail in https://github.com/arkworks-rs/poly-commit/pull/142, right? Or is there something else there broken?

**Comment by Pratyush** [comment]
Yup it should.

File: test-templates/src/groups.rs [hunk 1]
```diff
@@ -111,6 +111,18 @@ macro_rules! __test_group {
                         assert_eq!(context.mul_with_table(&bad_table, &b), None);
                     }
                 }
+
+                // The table size is only determined by the number of non-zero entries.
```
Comment (mmagician) [inline test-templates/src/groups.rs:115]: I think the table size could be anything the user sets.
E.g. [here](https://github.com/arkworks-rs/poly-commit/blob/f6a0c13bf44d7d1776f238bc9afabeab3c5a9511/poly-commit/src/multilinear_pc/mod.rs#L58) the table size is `num_vars`, while we have `2^(num_vars+1) - 1` scalars, if I'm reading the code correctly.

Comment (Pratyush) [reply test-templates/src/groups.rs:115]: In that case I'll rename `max_num_scalars` to just `num_scalars`.

**Review by mmagician — APPROVED** [review]
That is better yes
